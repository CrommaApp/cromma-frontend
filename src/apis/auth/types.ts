import { BasicResult } from '@apis/index';

type userInfo = {
	Keywords: string[];
	createdAt: string;
	id: number;
	updatedAt: string;
	userId: string;
};

export type SignRequest = {
	userId: string;
	password: string;
};
export type LoginResult = BasicResult & {
	data: userInfo;
};

export type getMyInfoResult = BasicResult & {
	data: userInfo | null;
};
