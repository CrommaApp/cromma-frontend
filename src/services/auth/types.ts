import { BasicResult } from '@services/api';
import { Post } from '@services/post/types';

type userInfo = {
	createdAt: string;
	id: number;
	updatedAt: string;
	userId: string;
};

export type SignRequest = {
	userId: string;
	password: string;
};
export type LoginResult = BasicResult & {
	data: userInfo;
};

export type getMyInfoResult = BasicResult & {
	data: userInfo | null;
};
