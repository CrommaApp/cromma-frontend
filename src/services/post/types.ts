import { BasicResult } from '@services/api';

export type Post = {
	id: number;
	title: string;
	content: string;
	createdAt: string;
	updatedAt: string;
};

export type GetAllPostsResult = BasicResult & {
	data: Post[] & { UserId: string };
};

export type UploadPostRequest = {
	title: string;
	content: string;
};

export type UploadPostResult = BasicResult & {
	data: Post;
};

export type getPostRequest = {
	postId: string;
};

export type deletePostRequest = {
	postId: string;
};
