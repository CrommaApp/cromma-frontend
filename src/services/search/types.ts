export type AddRecentKeywordRequest = {
	content: string;
};
