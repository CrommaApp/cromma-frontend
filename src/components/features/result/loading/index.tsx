import React from 'react';
import styled from 'styled-components';

const LoadingContainer = styled.div`
	display: flex;
	align-items: center;
	justify-content: center;
	width: 65%;
	height: 60%;
	margin-top: 5%;
	padding: 2%;
	background-color: white;
	border: 1px solid #dddddd;
	border-radius: 8px;

	@media screen and (max-width: 480px) {
		width: 80%;
		height: 70%;
	}
`;

const Loading = () => {
	return (
		<LoadingContainer>
			<h2>loading...</h2>
		</LoadingContainer>
	);
};

export default Loading;
