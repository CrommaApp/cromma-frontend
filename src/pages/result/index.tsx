import React from 'react';
import { useParams } from 'react-router';
import SearchResult from '@components/features/result/search-result';
import styled from 'styled-components';

const ResultContainer = styled.section`
	width: 100%;
	height: 100%;
	padding: 8% 2% 0 2%;
	display: flex;
	flex-direction: column;
	align-items: center;
	color: #888888;

	& > h1 {
		margin: 0;
		font-family: var(--font-sans-bold);
		font-size: 2.75rem;

		@media screen and (max-width: 768px) {
			font-size: 2.5rem;
		}

		@media screen and (max-width: 480px) {
			font-size: 2rem;
		}
	}
`;

const PageButtons = styled.div`
	width: 100%;
	padding: 3% 12%;
	position: absolute;
	bottom: 0;
	display: flex;
	justify-content: space-between;
	align-items: center;
	width: 100%;
	background-color: white;

	& > button {
		width: 20%;
		padding: 3% 0;
		font-size: 1.5rem;
		font-family: var(--font-sans-bold);
		border-radius: 100px;
		border: 1px solid #dddddd;

		&:first-child {
			background-color: white;
			color: #888888;
		}
		&:last-child {
			background-color: #dddddd;
			color: white;
		}

		@media screen and (max-width: 480px) {
			font-size: 1.2rem;
			width: 32%;
		}
	}
`;

type Params = {
	keyword: string;
};

const Result = () => {
	const params = useParams<Params>();

	return (
		<ResultContainer>
			<h1 id="search_result">Search results for {params.keyword}.</h1>
			<SearchResult />
			<PageButtons aria-label="pagination buttons">
				<button type="button">prev</button>
				<button type="button">next</button>
			</PageButtons>
		</ResultContainer>
	);
};

export default Result;
