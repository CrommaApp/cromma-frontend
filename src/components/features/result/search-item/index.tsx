import React from 'react';
import { News } from 'src/types/result/types';
import styled from 'styled-components';

const SearchItemContainer = styled.li`
	width: 100%;
	border: 1px solid #dddddd;
	border-radius: 8px;
	margin: 2% 0;
	padding: 3%;

	& > h2 {
		font-size: 1.5rem;
		margin: 0 0 1% 0;

		@media screen and (max-width: 768px) {
			font-size: 1.3rem;
		}
	}

	& > time {
		font-size: 0.75rem;
		color: #aaaaaa;
	}

	& > p {
		font-size: 1rem;
	}

	@media screen and (max-width: 480px) {
		padding: 4%;
	}
`;

type Props = {
	news: News;
};

const SearchItem = ({ news }: Props) => {
	return (
		<SearchItemContainer>
			<h2>{news.title}</h2>
			<time dateTime={news.wrt_dt}>{news.wrt_dt}</time>
			<p>{news.txt_origin_cn}</p>
		</SearchItemContainer>
	);
};

export default SearchItem;
