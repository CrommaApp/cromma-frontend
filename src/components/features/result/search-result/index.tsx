import React from 'react';
import styled from 'styled-components';
import SearchItem from '../search-item';
import { News } from 'src/types/result/types';

export const dummyNews: News[] = [
	{
		sfty_notice_id: '1',
		title: 'title',
		wrt_dt: '2021-12-14',
		txt_origin_cn: 'testtest',
	},
	{
		sfty_notice_id: '2',
		title: 'title',
		wrt_dt: '2021-12-14',
		txt_origin_cn: 'testtest',
	},
	{
		sfty_notice_id: '3',
		title: 'title',
		wrt_dt: '2021-12-14',
		txt_origin_cn: 'testtest',
	},
	{
		sfty_notice_id: '4',
		title: 'title',
		wrt_dt: '2021-12-14',
		txt_origin_cn: 'testtest',
	},
];

const SearchList = styled.ul`
	width: 65%;
	height: 60%;
	margin-top: 5%;
	padding: 2%;
	background-color: white;
	border: 1px solid #dddddd;
	border-radius: 8px;
	overflow-y: scroll;

	@media screen and (max-width: 480px) {
		width: 80%;
		height: 70%;
	}
`;

const SearchResult = () => {
	return (
		<SearchList aria-labelledby="search_result">
			{dummyNews.map((news, index) => (
				<SearchItem key={news.sfty_notice_id} news={news} index={index} />
			))}
		</SearchList>
	);
};

export default SearchResult;
