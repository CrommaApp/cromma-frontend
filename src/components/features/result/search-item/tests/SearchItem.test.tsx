import React from 'react';
import { render, screen } from '@testing-library/react';
import { News } from 'src/types/result/types';
import renderer from 'react-test-renderer';
import SearchItem from '@components/features/result/search-item';

describe('SearchItem', () => {
	const mockNews: News = {
		sfty_notice_id: '1',
		title: 'title',
		wrt_dt: '2021-12-14',
		txt_origin_cn: 'testtest',
	};

	beforeEach(() => {
		render(<SearchItem news={mockNews} index={1} />);
	});

	it('Snapshot', () => {
		const component = renderer.create(<SearchItem news={mockNews} index={1} />);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Show news information delivered through props', () => {
		expect(
			screen.getByRole('heading', {
				name: 'title',
			}),
		).toBeInTheDocument();

		expect(screen.getByText('testtest')).toBeInTheDocument();

		expect(screen.getByText('2021-12-14')).toBeInTheDocument();
	});
});
