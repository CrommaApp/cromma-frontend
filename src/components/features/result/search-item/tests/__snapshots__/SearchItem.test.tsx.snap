// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem Snapshot 1`] = `
<li
  aria-labelledby="news_title_1"
  className="sc-bdvvtL rSfIV"
>
  <h2
    id="news_title_1"
  >
    title
  </h2>
  <time
    dateTime="2021-12-14"
  >
    2021-12-14
  </time>
  <p>
    testtest
  </p>
</li>
`;
